lightning/src/ln/channelmanager.rs

```diff
@@ -779,6 +779,9 @@ macro_rules! handle_error {
 						});
 					}
 				}
+				if let Some(channel_id) = chan_id {
```

Comment:
Is there another way to detect whether we're actually closing the channel here before generating the event? Seems like the err could be an `IgnoreError` here, so it seems a bit error-prone to rely on whether `chan_id.is_some()` to generate the event

Reply:
Yeah locally this is one has already been moved inside the `if let Some((shutdown_res, update_option)) = shutdown_finish` just above!